mgmt_cli_test.py

```diff
@@ -0,0 +1,135 @@
+        stress_cmd = self.params.get('stress_cmd')
+        stress_cmd_queue = self.run_stress_thread(stress_cmd=stress_cmd)
+
+        self.log.info('Sleeping for 15s to let cassandra-stress run...')
+        time.sleep(15)
+        self.log.debug("test_mgmt_cli: initialize MgmtRepair nemesis")
+        self.db_cluster.add_nemesis(nemesis=MgmtRepair,
+                                    loaders=self.loaders,
+                                    monitoring_set=self.monitors,
+                                    )
+        self.db_cluster.start_nemesis()
+
+    def test_mgmt_cluster_crud(self):
+        """
+
+        Test steps:
+        1) add a cluster to manager.
+        2) update the cluster attributes in manager: name/host/ssh-user
+        3) delete the cluster from manager and re-add again.
+        """
+
+        manager_tool = mgmt.ScyllaManagerTool(manager_node=self.monitors.nodes[0])
+        hosts = self._get_cluster_hosts_ip()
+        selected_host = hosts[0]
+        cluster_name = 'mgr_cluster1'
+        mgr_cluster = manager_tool.add_cluster(name=cluster_name, host=selected_host)
+
+        # Test cluster attributes
+        cluster_orig_name = mgr_cluster.name
+        mgr_cluster.update(name="{}_renamed".format(cluster_orig_name))
+        assert mgr_cluster.name == cluster_orig_name+"_renamed", "Cluster name wasn't changed after update command"
+
+        # the below test currently fails: https://github.com/scylladb/mermaid/issues/741
+        # new_ssh_user="super-scylla-manager"
+        # mgr_cluster.update(ssh_user=new_ssh_user)
+        # assert mgr_cluster.ssh_user == new_ssh_user, "Cluster ssh-user wasn't changed after update command"
+
+        if len(hosts) > 1:
+            mgr_cluster.update(host=hosts[1])
+            assert mgr_cluster.host == hosts[1], "Cluster host wasn't changed after update command"
+        mgr_cluster.delete()
+        mgr_cluster2 = manager_tool.add_cluster(name=cluster_name, host=selected_host)
+
+    def _get_cluster_hosts_ip(self):
+        return [node_data[1] for node_data in self._get_cluster_hosts_with_ips()]
+
+    def _get_cluster_hosts_with_ips(self):
+        ip_addr_attr = 'public_ip_address' if self.params.get('cluster_backend') != 'gce' and \
+                                              len(self.db_cluster.datacenter) > 1 else 'private_ip_address'
+        return [[n, getattr(n, ip_addr_attr)] for n in self.db_cluster.nodes]
+
+    def test_manager_sanity(self):
+        """
+        Test steps:
+        1) Run the repair test.
+        2) Run test_mgmt_cluster test.
+        :return:
+        """
+
+        self.test_mgmt_repair_nemesis()
+        self.test_mgmt_cluster()
+
+    def test_mgmt_cluster_healthcheck(self):
+
+        manager_tool = mgmt.ScyllaManagerTool(manager_node=self.monitors.nodes[0])
+        selected_host_ip = self._get_cluster_hosts_ip()[0]
+        mgr_cluster = manager_tool.add_cluster(name='mgr_cluster1', host=selected_host_ip)
```

Comment:
Do we want to handle the case that a cluster with that name already exist (command may fail)